### PR TITLE
[URGENT] Left navigation format fixes - "_files" exclusion and root level fixes

### PR DESCRIPTION
**Pull request by kaylaziegler**
Trello story: https://trello.com/c/zwuWdHeF

Issue #265 

**Dev Cascade**
- Test formats: `_components/left_nav/`
- Live formats: `_cascade/formats/level/`

### CODE REVIEW

**Comment by kaylaziegler** [comment]
See issue #265 for the explanation about why the `left_nav_no_siblings` format didn't have the problem with the title when the format was used directly under root.

Only thing I had to change in that one was the other part of this, which was to exclude folders called `_files`.

File: .cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl [hunk 1]
```diff
@@ -25,14 +25,25 @@
...
 #macro ( buildLeftNavTitle )
     ## Data
-    #set ( $folderIndexPage = $_.locatePage("${parentDir.path}/index" ) )
+    #if ( $parentDir.path == "/" )
+        ## For folders under the root, the index doesn't need another forward slash
+        #set ( $folderIndexPage = $_.locatePage("${parentDir.path}index" ) )
+        #set ( $titleLink = $folderIndexPage.link )
+        ## This $title variable should be $folderIndexPage.label, but for the home page
+        ## that label is way too long.
+        #set ( $leftNavTitle = "Chapman University" )
```
Comment (kaylaziegler) [inline .cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl]: Discussed this with Ross. The value here is technically supposed to be "Chapman University | A Top Private University in California" because that's the display name for the home page.

That's just a bit too wordy, so Ross said "Chapman University" for the title if the parent of the current page is root.

File: .cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl [hunk 2]
```diff
@@ -50,12 +61,13 @@
...
         ## Sibling is a folder
-        #elseif ( $sibling.assetType == 'folder' && $isVisible )
+        #elseif ( $sibling.assetType == 'folder' && $isVisible && $sibling.name != '_files' )
             #set ( $indexPage = $_.locatePage("${sibling.path}/index") )
             #set ( $visibleIndex = $indexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
             
             ## Only if the index page is also visible
-            #if ( $visibleIndex )
+            ## Also make sure the correct index page was found
+            #if ( $visibleIndex && ( $indexPage.path == "${sibling.path}/index" ) )
```
Comment (kaylaziegler) [inline .cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl]: These last two commits are to deal with folders that don't have index pages for whatever reason. They cause problems similar to what we saw in issue #262 

This is the most effective way I think to do it since velocity doesn't return anything helpful like NULL when it doesn't locate something, it leaves it as the previous value of that variable.

File: .cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl [hunk 3]
```diff
@@ -102,12 +103,13 @@
...
         #if ( $asset.assetType == 'folder' && $asset.name != '_files' )
-            ## Both the folder and index page must be visible
             #set ( $folderVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
             #set ( $folderIndexPage = $_.locatePage("${asset.path}/index" ) )
             #set ( $folderIndexVisible = $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
             
-            #if ( $folderIndexVisible && $folderVisible )
+            ## Both the folder and index page must be visible
+            ## Also make sure this asset's index page was found correctly
+            #if ( $folderIndexVisible && $folderVisible && ( $folderIndexPage.path == "${asset.path}/index" ) )
```
Comment (tatwell) [inline .cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl]: How about this approach to explain what `$folderIndexPage.path == "${asset.path}/index"` means.

```
              #set ( $folderVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
              #set ( $folderIndexPage = $_.locatePage("${asset.path}/index") )
              #set ( $folderIndexPageExists = $folderIndexPage.path == "${asset.path}/index" )
              #set ( $folderIndexVisible = $folderIndexPageExists && $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )

              #if ( $folderVisible && $folderIndexVisible )
```

You'd have to change it a couple other places, too.
